### PR TITLE
New PR to add Elevate24 

### PR DESCRIPTION
**Pull request by liarrmo**
Hey, 

Creating a pull request to add Elevate24 to Manifests. Let me know if I need to do anything. First time so apologies in advance! 

Thanks

### CODE REVIEW

**Review by relgit — CHANGES_REQUESTED** [review]
Thanks for the submission @liarrmo. The work you did is already pretty extensive; I just have a handful of requests:

* Apart for the properties from the template, none have default values. This is especially important for boolean ones, but is also useful for others. Please add values using `pfm_default`.
* If a text field has a default value, the placeholder should normally be equal to it. For example, `showAdminPasswordGrace` has a default of `0` so its placeholder should be `0` too.
* Continuing with `showAdminPasswordGrace`, please add a `pfm_range_max` tag equal to `45`.
* Please include a separate icon file in `Icons/ManagedPreferencesApplications/` instead of in `pfm_icon`.
* And lastly, please run the entire manifest through the following command. This will standardize the XML and remove comments: `plutil -convert xml1 com.jigsaw24.Elevate24.plist`.

**Comment by liarrmo** [comment]
Thank you for the feedback! I’ve resolved all the below, so I will get the changes over. I wanted to ask, we have one bit of configuration which is configured as

<key>microsoftsentinel</key>
<array>
<dict>
<key>SentinelClientId</key>
<string>ClientID</string>
<key>SentinelTennantId</key>
<string>TennantID</string>
<key>SentinelClientsecret</key>
<string>Secret</string>
<key>SentinelUploadURL</key>
<string>URL</string>
</dict>
</array>

Is this possible to represent in the manifest? I can’t see how to do it.

Thanks!


Liam Donnelly | Technical Lead for Workplace Solutions
T: 07711 378 220 | E: ***@***.*** 

L: 
linkedin.com/Jigsaw24 | 
Y: 
youtube.com/WeAreJigsaw24


​​​Please consider the environment - do you really need to print this email?
​​​
​This email may contain confidential information, personal data and/or copyright material and is intended for the use of the addressee only. If you are not the addressee any unauthorised use may be unlawful. If you receive this email by mistake, please advise the sender immediately by using the reply facility in your email software. Any views or opinions presented are those of the sender and do not necessarily represent Jigsaw Systems Limited (trading as Jigsaw24). Jigsaw Systems Limited (2682904) is registered in England and Wales. For further information on our privacy policy please refer to our website. ​Jigsaw24 have taken steps to ensure that this email message and any attachments have been scanned for the presence of viruses, we advise that in keeping with good computing practice the recipient should carry out checks to ensure that they are free of viruses.
 
From: Ari Leviatan ***@***.***>
Date: Wednesday, 25 September 2024 at 08:36
To: ProfileCreator/ProfileManifests ***@***.***>
Cc: Liam Donnelly ***@***.***>, Mention ***@***.***>
Subject: Re: [ProfileCreator/ProfileManifests] New PR to add Elevate24 (PR #708)

CAUTION: This email originated from outside the organisation. Do not click links or open attachments unless you recognise the sender and know the content is safe.

@relgit requested changes on this pull request.

Thanks for the submission @liarrmo<https://github.com/liarrmo>. The work you did is already pretty extensive; I just have a handful of requests:

  *   Apart for the properties from the template, none have default values. This is especially important for boolean ones, but is also useful for others. Please add values using pfm_default.
  *   If a text field has a default value, the placeholder should normally be equal to it. For example, showAdminPasswordGrace has a default of 0 so its placeholder should be 0 too.
  *   Continuing with showAdminPasswordGrace, please add a pfm_range_max tag equal to 45.
  *   Please include a separate icon file in Icons/ManagedPreferencesApplications/ instead of in pfm_icon.
  *   And lastly, please run the entire manifest through the following command. This will standardize the XML and remove comments: plutil -convert xml1 com.jigsaw24.Elevate24.plist.

**Comment by relgit** [comment]
Yes. The following is a stripped-down example you can build on. It will appear as a multi-column table.
```xml
...
<dict>
    <key>pfm_name</key>
    <string>microsoftsentinel</string>
    <key>pfm_subkeys</key>
    <array>
        <dict>
            <key>pfm_name</key>
            <string>microsoftsentinelItem</string>
            <key>pfm_subkeys</key>
            <array>
                <dict>
                    <key>pfm_name</key>
                    <string>SentinelClientId</string>
                    <key>pfm_type</key>
                    <string>string</string>
                </dict>
                <dict>
                    <key>pfm_name</key>
                    <string>SentinelTennantId</string>
                    <key>pfm_type</key>
                    <string>string</string>
                </dict>
                ...
            </array>
            <key>pfm_type</key>
            <string>dictionary</string>
        </dict>
    </array>
    <key>pfm_type</key>
    <string>array</string>
</dict>
...
```

**Comment by liarrmo** [comment]
Appreciate it, thank you! I've made all the changes and looks good from my end. Let me know if I have missed anything off?

**Comment by relgit** [comment]
Thanks for making the requested changes.

Looks like you followed all of the advice, but there's a small confusion about value types. These should be of the same type as that of the property. This applies to values of `pfm_default` and `pfm_range_max` so for example, a boolean property default value could be `<false/>` or `<true/>`, and an integer's `<integer>45</integer>`.

You did mention it was your first manifest 😀

By the way, this does not apply to `pfm_value_placeholder` which represents text and therefore must always be `string`.

**Comment by liarrmo** [comment]
Well that makes perfect sense now! Thank you so much for your help with this, I've made those changes now

File: Manifests/ManagedPreferencesApplications/com.jigsaw24.Elevate24.plist [hunk 1]
```diff
@@ -0,0 +1,874 @@
...
+			<key>pfm_default</key>
+			<string>Elevate24</string>
+			<key>pfm_description</key>
+			<string>Change the standard button text from Elevate in the application. </string>
+			<key>pfm_documentation_url</key>
+			<string>https://github.com/Jigsaw24/Elevate24/blob/main/Documentation/Elevate24%20-%20Configuration%20Keys.md#branding</string>
+			<key>pfm_name</key>
+			<string>mainbutton</string>
+			<key>pfm_title</key>
+			<string>Mainbutton Txt (Premium)</string>
```
Comment (relgit) [inline Manifests/ManagedPreferencesApplications/com.jigsaw24.Elevate24.plist:385]: Let's change "Mainbutton Txt" to "Main Button Text".

File: Manifests/ManagedPreferencesApplications/com.jigsaw24.Elevate24.plist [hunk 2]
```diff
@@ -0,0 +1,874 @@
...
+			<key>pfm_title</key>
+			<string>Password Symbols (Premium)</string>
+			<key>pfm_type</key>
+			<string>string</string>
+			<key>pfm_value_placeholder</key>
+			<string>2</string>
+		</dict>
+		<dict>
+			<key>pfm_default</key>
+			<string>0</string>
```
Comment (relgit) [inline Manifests/ManagedPreferencesApplications/com.jigsaw24.Elevate24.plist:846]: This value should also be of type integer.

**Review by relgit — CHANGES_REQUESTED** [review]
Alright last two comments and we're done.

**Comment by liarrmo** [comment]
Thanks, good spot. All sorted. Appreciate it!

**Pull request merged**